### PR TITLE
rough implementation of roscontrol associated packages

### PR DESCRIPTION
**Pull request by wmmc88**

### CODE REVIEW

**Review by matthew-reynolds — DISMISSED** [review]
Handful of remaining Rover things (pkg desc, yaml, transmisssions) and other stuff I would normally ask for changes on, but fuck that :P just gotta make it walk

File: qpup_hw/config/qpup_hardware.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+qpup:
+  combined_robot_hw:
+    shared_values:
+      splay_transmission_reduction: &shared_splay_transmission_reduction 40
+      hip_transmission_reduction: &shared_hip_transmission_reduction 40
+      knee_transmission_reduction: &shared_knee_transmission_reduction deg(rad(40 * 63 / 39))
```
Comment (matthew-reynolds) [inline qpup_hw/config/qpup_hardware.yaml:6]: Why `deg(rad(...))`, just to force an evaluation? Also does this int division truncate? 96.9230769 -> 96 is not good

File: qpup_hw/config/qpup_hardware.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    qpup_base:
+      type: qpup_hw/UQPUPHWReal
```
Comment (matthew-reynolds) [inline qpup_hw/config/qpup_hardware.yaml]:
```suggestion
      type: qpup_hw/QPUPHWReal
```

File: qpup_hw/config/qpup_hardware.yaml (hunk 1 above)
Comment (wmmc88) [reply qpup_hw/config/qpup_hardware.yaml:6]: just to force the evaluation. it def evaluates as float tho. the contents is evaluated in python

**Review by matthew-reynolds — APPROVED** [review]

**Pull request merged**